Add config for disabling use of root network namespace

hmahmood opened the pull request:
<!--
* New contributors are highly encouraged to read our
  [CONTRIBUTING](/CONTRIBUTING.md) documentation.
* The pull request:
  * Should only fix one issue or add one feature at a time.
  * Must update the test suite for the relevant functionality.
  * Should pass all status checks before being reviewed or merged.
* Commit titles should be prefixed with general area of pull request's change.
* Draft PRs should be prefixed with `[WIP]` in their title.

-->
### What does this PR do?
Adds a new config `system_probe_config.enable_root_netns` to enable/disable use of the root network namespace.

<!--
* A brief description of the change being made with this pull request.
* If the description here cannot be expressed in a succint form, consider
  opening multiple pull requests instead of a single one.
-->

### Motivation

<!--
* What inspired you to submit this pull request?
* Link any related GitHub issues or PRs here.
-->

### Additional Notes

<!--
* Anything else we should know when reviewing?
* Include benchmarking information here whenever possible.
* Include info about alternatives that were considered and why the proposed
  version was chosen.
-->

### Possible Drawbacks / Trade-offs

<!--
* What are the possible side-effects or negative impacts of the code change?
-->

### Describe how to test/QA your changes
QA instructions here for testing against a test in the code base, so you will have to sync to the release branch being tested on a linux VM.
1. From the top of the repo, run: ```go test -tags linux_bpf ./pkg/network/config -c```; this will create a test binary `config.test` 
2. create a new network namespace: ```sudo ip netns add test```
3. run the test in the new namespace: ```sudo ip netns exec test ./config.test -test.run TestDisableRootNetNamespace -test.v -test.count 1```
4. cleanup network namespace: ```sudo ip netns del test```

### Reviewer's Checklist
<!--
* Authors can use this list as a reference to ensure that there are no problems
  during the review but the signing off is to be done by the reviewer(s).

Note: Adding GitHub labels is only possible for contributors with write access.
-->

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

hmahmood committed:
Add config for disabling use of root network namespace

brycekahle commented on pkg/config/system_probe.go:
```diff
@@ -165,6 +165,9 @@ func InitSystemProbeConfig(cfg Config) {
 	cfg.BindEnvAndSetDefault(join(smNS, "enabled"), false, "DD_SYSTEM_PROBE_SERVICE_MONITORING_ENABLED")
+
+	// enable/disable use of root net namespace
+	cfg.BindEnvAndSetDefault(join(netNS, "disable_root_netns"), false)
```
Can we flip it to an enable flag, with the default being true? I find it easier to read that way.

brycekahle replied:
Maybe name it `use_root_netns`?

brycekahle commented on pkg/network/config/config.go:
```diff
@@ -221,6 +237,8 @@ func New() *Config {
 
 		RecordedQueryTypes: cfg.GetStringSlice(join(netNS, "dns_recorded_query_types")),
+
+		DisableRootNetNs: cfg.GetBool(join(spNS, "disable_root_netns")),
```
```suggestion
		DisableRootNetNs: cfg.GetBool(join(netNS, "disable_root_netns")),
```

brycekahle reviewed (approved):
This looks good. Any ideas on how we could test this to ensure we don't break the option in the future?

hmahmood committed:
Overlooked instances of root namespace being used

hmahmood committed:
Add test

hmahmood commented:
> This looks good. Any ideas on how we could test this to ensure we don't break the option in the future?

I added a very basic test, but without being able to run the test in a different process or network namespace, we can't add an automated test. The test I added can be run manually in a different namespace, so I'll add instructions for that for qa.

hmahmood committed:
Remove net ns from config

hmahmood committed:
Fix windows build

hmahmood committed:
Merge remote-tracking branch 'origin/main' into hasan.mahmood/disable-root-ns
